File: Product_Management/src/main/java/com/product/model/service/productService.java
package com.product.model.service;

import java.util.List;

import com.product.model.modelentity.product;

public interface productService {
	
	public product saveProduct(product products);
	
	public List<product> getAllProduct();
	
	public product getProductById(Integer id);
	
	public String deleteProduct(Integer id);
	
	public product editProduct(product products, Integer id);
	
	

}

File: product_ui/src/service/product.service.js
import App from "../App";
import axios from "axios";

const API_URL= "http://localhost:8080"

class ProductService{
    saveProduct(products){
        return axios.post(API_URL+"/saveProduct", products)
    }

    getAllProduct(){
        return axios.get(API_URL+"/");

    }
    getAllProductById(id){
        return axios.get(API_URL+"/"+id);  
    }

    deleteProduct(id){
        return axios.get(API_URL+"/deleteProduct/"+id);

    }

    editProduct(products){
        return axios.post(API_URL+"/"+products.id,products);
    }

}

export default new ProductService;